pkg/services/k8s/apiserver/service.go

```diff
@@ -47,7 +50,9 @@ type service struct {
 	stoppedCh chan error
 }
 
-func ProvideService(etcdProvider kine.EtcdProvider, cfg *setting.Cfg) (*service, error) {
+// NOTE: k8sAuthnAPI and k8sAuthzAPI aren't code-dependencies of apiserver. However, they need to be exercised somewhere
+// in order for Wire to pull them in. These can be moved to a better place, if there is such a candidate.
+func ProvideService(etcdProvider kine.EtcdProvider, _ authnz.K8sAuthnAPI, _ authnz.K8sAuthzAPI, cfg *setting.Cfg) (*service, error) {
```

Comment:
what do you think about adding `NewIdleService` to `K8sAuthnAPIImpl` and `K8sAuthnAPIImpl`  and moving `RegisterAPIEndpoints()` in each of them to the `start` functions? that way we can avoid registering API routes if the k8s apiserver is not enabled

Reply:
discussed in slack, and this doesn't need to be addressed in this PR